doc/development_guide/api/index.rst

```diff
@@ -0,0 +1,10 @@
+###
+API
+###
+
```

Comment:
We should probably write something here. https://pylint--6658.org.readthedocs.build/en/6658/development_guide/api/index.html feels a bit empty.